interfaces/prompting/constraints.go

```diff
@@ -97,9 +164,9 @@ func (c *Constraints) Match(path string) (bool, error) {
 	return match, nil
 }
 
-// ContainPermissions returns true if the constraints include every one of the
-// given permissions.
-func (c *Constraints) ContainPermissions(permissions []string) bool {
+// ContainPermissions returns true if the permission list in the constraints
+// includes every one of the given permissions.
+func (c *ReplyConstraints) ContainPermissions(permissions []string) bool {
```

Comment:
iiuc ReplyContraints is not really validated unless via ToConstraints which makes me think that we really want ReplyContraints to not have any methods except the conversion and move the methods even if annoying to Constraints. That should force changing some of the code I'm right now slightly uncomfortable with.

Reply:
I think that makes sense: `ReplyConstraints.ContainPermissions` and `ReplyConstraints.Match` can be moved to methods on `Constraints` instead.

The rationale for having those methods on `ReplyConstraints` instead of `Constraints` is that one should never call such methods on any constraints except ones which are received as part of a reply. But we can add a comment to that effect instead.

Reply:
You were also right about moving `responseForInterfaceConstraintsOutcome` in #14390, it's making the rebase much more difficult...